src/config.h

```diff
@@ -338,7 +338,7 @@ void setcpuaffinity(const char *cpulist);
 #define HAVE_FADVISE
 #endif
 
-#define IO_THREADS_MAX_NUM 16
```

Comment:
Was the number `16` decided earlier for users to avoid shooting themselves in the foot i.e. performance will start deteriorating beyond that count?

Reply:
IIRC, we use 3 bits in an robj to store which thread allocated an robj, so it can be passed back to be freed by the same thread.

Reply:
So, we can't blindly increase this?

Reply:
Maybe not. :) Can you look it up?

There was some discussion related to the PR #763.

Reply:
> Was the number `16` decided earlier for users to avoid shooting themselves in the foot i.e. performance will start deteriorating beyond that count?

@hpatro Per my testing before, we will observe a significant performance degradation when the io-threads number greater than online CPUs, this issue still exists with the new valkey async IO, the spinning io-threads preempt CPU from main thread. It should have no relationship with the `16` number. I remember @madolson suggested to trust the operators :). https://github.com/redis/redis/issues/13003

The number `16` @uriyage responded in https://github.com/valkey-io/valkey-private/pull/6#discussion_r1597652783 before, he couldn't observe any improvement beyond 8-10 threads anyway.  Not sure if he tested the larger payloads (in MBs).

Reply:
The work to limit the number of IO threads was so that we could send free requests to the same thread that allocated as was mentioned. For various reasons we decided to not commit to that as part of the V1, but I believe it's on one of the follow items: https://github.com/valkey-io/valkey/issues/761. 

I do believe in trusting operators/admins. (Never trust application developers though, they shoot themselves in the foot). We should definitely be "sane" by default, but give power users the operational tools to do advanced things.